docs/reference/runtime/runtime-class.md

```diff
@@ -77,7 +77,77 @@ runtime.registerRoutes([
 ]);
 ```
 
-[!ref icon="gear" text="Setup the host application to accept hoisted routes"](/reference/routing/useHoistedRoutes.md)
+[!ref text="Setup the host application to accept hoisted routes"](/reference/routing/useHoistedRoutes.md)
+
+### Register routes under a specific nested layout route
+
+React router [nested routes](https://reactrouter.com/en/main/start/tutorial#nested-routes) enable applications to render nested layouts at various points within the router tree. This is quite helpful for federated applications as it enables composable and decoupled UI.
+
+To fully harness the power of nested routes, the `registerRoutes` function allows a route to be registered **under any** previously registered **nested layout route**, even if that route was registered by another module.
+
+When registering a new route with the `registerRoutes` function, to render the route under a specific nested layout route, specify a `layoutPath` property that matches the nested layout route's `path` property. The only requirement is that the **nested layout route** must has been **registered** to `@squide` **before** the **new child route**.
+
+```tsx !#10
+import { lazy } from "react";
+
+const Page = lazy(() => import("./Page.tsx"));
+
+runtime.registerRoutes([
+    {
+        path: "/layout/page-1",
+        element: <Page />
+    }
+], { layoutPath: "/layout" }); // Register the page under the "/layout" nested layout.
+```
+
+!!!info
+Likewise any other React Router routes, the `path` property of a page rendered under a nested layout must be an absolute path. For example, if a nested layout `path` is `/layout`, the `path` property of a page rendered under that layout route and responding to the `/page-1` url, should be `/layout/page-1`.
+!!!
+
+#### Index routes
+
+Although nested layout routes that serves as indexes (e.g. `{ index: true, element: <Layout /> }`) are not very common, `@squide` still supports this scenario. To register a route **under an index route**, set the `layoutPath` property as the concatenation of the index route's parent path and `/$index$`. 
```

Comment:
/$index$ seems like black magic to me. I'm not sure why we need nested routes to know that the parent is an index route either

Reply:
Do we have to be bound to the path? Could we not register the layout with a special key and then register the nested route with the same key?

Reply:
We could but a `path` is already defined for the nested layout route and must be reused by every module that are registering routes under the nested layout to define those routes `path` property.

Example:

If the nested layout path is `/layout` and a module is registering under that nested layout the page `/page-1`, the page `path` must be  `/layout/page-1`  (this is how React Router works).

Therefore, since every module must already know the nested layout `path` and by nature a path  (url) is a unique key, IMO, adding another key complicates thing more than it helps.

Reply:
I guess in any case we'll probably have a package to share layout paths between modules, so I guess either way it doesn't make a ton of difference.

My only worry is that if someone removes `index: true` from a layout path they probably won't expect that nested pages won't bind to it anymore, so might cause bugs.

Reply:
Indeed that's an issue but there's a validation to check that and an error will be thrown if the layout route is not found. Still, that might be problematic as it will only be discovered at integration.

I agree that there might be something here to improve here but I am not sure what and how. I would vote to inspect how it goes and adapt if needed. What do you think?

Reply:
Yeah, I don't think we use index paths a lot anyway, so we can leave as-is for now, it's not a blocker.